File: euler/problem-2/main.rb
def sum_of_even_fibo(limit)
  a, b = 1, 2
  total = 0
  while a <= limit
    total += a if a.even?
    a, b = b, a + b
  end
  return total
end

limit = 4000000
res = sum_of_even_fibo(limit)
puts res

File: euler/problem-2/main.cpp
#include <iostream>

int main(void) {
    int limit = 4000000;
    int total = 0;
    int a = 1;
    int b = 2;

    while (a <= limit) {
        if (a % 2 == 0) {
            total += a;
        }
        int temp = a + b;
        a = b;
        b = temp;
    }

    std::cout << total << std::endl;
    return 0;
}